core/types/receipt.go

```diff
@@ -572,124 +548,27 @@ func (r Receipts) DeriveFields(hash libcommon.Hash, number uint64, txs Transacti
 			logIndex++
 		}
 	}
-	return nil
-}
-
-func (r Receipts) ToReceiptsEncodable() ReceiptsEncodable {
-	receiptsEncodable := make([]*ReceiptEncodable, len(r))
-	for i, receipts := range r {
-		var (
-			l1GasPrice []byte
-			l1GasUsed  []byte
-			l1Fee      []byte
-			feeScalar  []byte
-			l2BobaFee  []byte
-			err        error
-		)
-		if receipts.L1GasPrice != nil {
-			l1GasPrice, err = (*receipts.L1GasPrice).GobEncode()
-			if err != nil {
-				log.Warn("failed to encode L1GasPrice", "err", err)
-			}
-		}
-		if receipts.L1GasUsed != nil {
-			l1GasUsed, err = (*receipts.L1GasUsed).GobEncode()
-			if err != nil {
-				log.Warn("failed to encode L1GasUsed", "err", err)
-			}
-		}
-		if receipts.L1Fee != nil {
-			l1Fee, err = (*receipts.L1Fee).GobEncode()
-			if err != nil {
-				log.Warn("failed to encode L1Fee", "err", err)
+	if config.IsOptimismBedrock(number) && len(txs) >= 2 { // need at least an info tx and a non-info tx
+		if data := txs[0].GetData(); len(data) >= 4+32*8 { // function selector + 8 arguments to setL1BlockValues
+			l1Basefee := new(uint256.Int).SetBytes(data[4+32*2 : 4+32*3]) // arg index 2
```

Comment:
So, basically, because we know this data must be in the deposit TX, we don't bother trying to retrieve it from the stateDB?  I like this a lot (and very happy to see the state transition hack around it disappear).

Reply:
> So, basically, because we know this data must be in the deposit TX, we don't bother trying to retrieve it from the stateDB? I like this a lot (and very happy to see the state transition hack around it disappear).

Yes! There is no hack any more.